### PR TITLE
delete the added user so that the scenario can call repeatedly

### PR DESCRIPTION
**Pull request by astridx**
### Summary of Changes

When I ran ./tests/codeception/vendor/bin/codecept run ./tests/codeception/acceptance/users.feature twice, I got a failure. 
The reason for this failure is, that in the first run an user is created and this user is not deleted. It is renamed, but the email address remains the same. So I had the idea to delete this user in the end of the scenario. 
I add Scenario: 
Delete user 
Given I have a user with user name "User Three" 
When I Delete the user "User Three" 
Then I should see "No Matching Results" for deleted user "User Three" 

to the end of my local file users_frontend.feature.

All respective methods are already in the step object.

### CODE REVIEW

File: tests/codeception/acceptance/users_frontend.feature [hunk 1]
```diff
@@ -50,3 +50,8 @@ Feature: Users Frontend
     And User "User Three" did login at least once
     When I login as a super admin from backend
     Then I should see last login date for "User Three"
+
+Scenario: 
+	Delete user Given I have a user with user name "User Three" 
+	When I Delete the user "User Three" 
+	Then I should see "No Matching Results" for deleted user "User Three"
```
Comment (nibra) [inline tests/codeception/acceptance/users_frontend.feature]: Small typo: Should be

``` Gherkin
Scenario: Delete user
    Given I have a user with user name "User Three" 
    When I Delete the user "User Three" 
    Then I should see "No Matching Results" for deleted user "User Three" 
```

Please remember the newline at the end of the file!

**Review by nibra — CHANGES_REQUESTED** [review]
Thank you, Astrid! Just those little corrections, and this can be merged.

**Comment by astridx** [comment]
Is the code style now OK?

**Review by nibra — APPROVED** [review]

**Review by nibra — APPROVED** [review]
Ready to merge for me.

**Comment by astridx** [comment]
I hope I'm not annoying with my ideas. If you are all sure that the tests should be repeatable, I would change the following, too:
Within the class ../tests/codeception/acceptance/administrator/UserCest.php
I would not only use Faker for the username like is is:

```
public function __construct()
    {
        $this->faker    = Faker\Factory::create();
        $this->name     = 'User' . $this->faker->randomNumber();
        $this->username = 'uname' . $this->faker->randomNumber();
        $this->email    = 'test@joomla.org';
        $this->password = 'test';
    }

```

I would use it for the email, too:

```
public function __construct()
    {
        $this->faker    = Faker\Factory::create();
        $this->name     = 'User' . $this->faker->randomNumber();
        $this->username = 'uname' . $this->faker->randomNumber();
        $this->email    = 'test' . $this->faker->randomNumber() . @joomla.org';
        $this->password = 'test';
    }
```

If you like it, I will add it to this PR.

**Comment by yvesh** [comment]
@astridx To be honest the test data should probably be static, in order to be really repeatable (and always have the same data) and not generated by faker.

The downside is that running tests on the same install (or single tests) would be impossible that way. This is the reason we currently randomized it. Having the mail also randomized for the later reasons is a good idea. Thank you!

**Comment by astridx** [comment]
In this case we should not use Faker. We should use a static user name and delete this user in the end of the test. Are I am right?

**Comment by puneet0191** [comment]
@astridx Please use Faker even for generating email's. I don't see any problem in that, also creating a user with Faker and then deleting the user in the end of the Tests should be the right way.

**Comment by astridx** [comment]
@Puneet Kala Thank you for your reply. If we delete the user in the end, we do not need Faker. Because then the tests are without Faker repeatable. 
Should I add deleting of the user in the end of the file ../tests/codeception/acceptance/administrator/UserCest.php and make the user name static?

**Comment by astridx** [comment]
@Puneet Thank you very much. After downgrade firefox everything worked well again.

I made a new PR for the UserCest.php: https://github.com/joomla-projects/gsoc16_browser-automated-tests/pull/83

**Pull request closed without merging**